crates/js_api/src/gui/state_management.rs

```diff
@@ -32,9 +32,9 @@ struct SerializedGuiState {
 
 #[wasm_export]
 impl DotrainOrderGui {
-    fn get_dotrain_hash(dotrain: String) -> Result<String, GuiError> {
-        let dotrain_bytes = bincode::serialize(&dotrain)?;
-        let hash = Sha256::digest(dotrain_bytes);
+    pub fn get_dotrain_hash(dotrain: String) -> Result<String, GuiError> {
```

Comment:
if we make this pub, we need to add `#[wasm_export(skip)]` so that it does not show up on js side